### PR TITLE
update smoldot

### PR DESCRIPTION
**Pull request by qiweiii**
https://github.com/AcalaNetwork/chopsticks/issues/754

update smoldot to include ext_storage_proof_size_storage_proof_size_version_1

### CODE REVIEW

File: executor/src/task.rs [hunk 1]
```diff
@@ -142,6 +142,8 @@ pub async fn run_task(task: TaskCall, js: crate::JsCallback) -> Result<TaskRespo
             storage_main_trie_changes,
             max_log_level: task.runtime_log_level,
             calculate_trie_changes: false,
+            storage_proof_size_behavior:
+                runtime_call::StorageProofSizeBehavior::ConstantReturnValue(0),
```
Comment (xlc) [inline executor/src/task.rs:146]: better to make this configurable from js side

Comment (qiweiii) [reply executor/src/task.rs:146]:
```rs
storage_proof_size_behavior:
         runtime_call::StorageProofSizeBehavior::ConstantReturnValue(task.storage_proof_size),
```
sth like this?

File: packages/core/src/blockchain/index.ts [hunk 2]
```diff
@@ -49,6 +49,11 @@ export interface Options {
   maxMemoryBlockCount?: number
   /** Whether to process queued messages */
   processQueuedMessages?: boolean
+  /**
+   * Storage Proof Size passed to runtime.
+   * See <https://github.com/smol-dot/smoldot/blob/26dedea00eafa7073ca1d4678277272d807b3d89/lib/src/executor/host.rs#L248>
+   */
+  storageProofSize?: number
```
Comment (xlc) [inline packages/core/src/blockchain/index.ts:56]: no need to put it here. this should be a per call option and defaults to 0 and for now the only way to set it to some value is from using the API directly

Comment (qiweiii) [reply packages/core/src/blockchain/index.ts:56]: which api do you refer to here? and how do user pass in this value

File: packages/core/src/wasm-executor/index.ts [hunk 3]
```diff
@@ -113,16 +122,10 @@ export const createProof = async (nodes: HexString[], updates: [HexString, HexSt
...
-    wasm: HexString
-    calls: [string, HexString[]][]
-    mockSignatureHost: boolean
-    allowUnresolvedImports: boolean
-    runtimeLogLevel: number
-  },
-  callback: JsCallback = emptyTaskHandler,
-) => {
+export const runTask = async (task: TaskCall, callback: JsCallback = emptyTaskHandler) => {
+  if (!task.storageProofSize) {
```
Comment (xlc) [inline packages/core/src/wasm-executor/index.ts:126]:
```js
let task2 = {
  ...task,
  storageProofSize: task.storageProofSize ?? 0
}
```

avoid modify input parameter

**Pull request merged**